File: app/src/main/java/com/example/dietiestates25/controller/AuthManager.java
package com.example.dietiestates25.controller;

public class AuthManager {
    private static AuthManager instance;
    private String idToken;

    private String role;

    private AuthManager() {}

    public static synchronized AuthManager getInstance() {
        if (instance == null) {
            instance = new AuthManager();
        }
        return instance;
    }

    public String getIdToken() {
        return idToken;
    }

    public void setIdToken(String token) {
        this.idToken = token;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }
}

File: app/src/main/java/com/example/dietiestates25/controller/AuthManager.kt
package com.example.dietiestates25.controller

import android.util.Log
import com.amplifyframework.auth.cognito.AWSCognitoAuthSession
import com.amplifyframework.core.Amplify

object AuthManager {
    var role: String? = null

    fun getToken(callBack: (String?) -> Unit) {
        Amplify.Auth.fetchAuthSession(
            { session -> if (session.isSignedIn) {
                val idToken = (session as AWSCognitoAuthSession).userPoolTokensResult.value?.idToken
                Log.i("AuthQuickstart", "id Token: $idToken")
                callBack(idToken)
            }},
            {
                error -> Log.e("Auth", "Failed to fetch auth session", error)
                callBack(null)
            }
        )
    }

}
